core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorUtil.java

```diff
@@ -1287,6 +1287,20 @@ private static FlatAggregate flattenRecurse(@Nullable SqlCall filterCall,
     }
   }
 
+  /** Returns whether a select item is a measure. */
+  public static boolean isMeasure(SqlNode selectItem) {
+    return getMeasure(selectItem) != null;
+  }
+
+  /** Returns the measure expression if a select item is a measure, null
+   * otherwise.
+   *
+   * <p>For a measure, {@code selectItem} will have the form
+   * {@code AS(MEASURE(exp), alias)} and this method returns {@code exp}. */
+  public static @Nullable SqlNode getMeasure(SqlNode selectItem) {
+    return null;
```

Comment:
This method seems unfinished, will it be part of future work? Maybe add a TODO or a comment in that case?

Reply:
Yes, it will be extended in [CALCITE-4496](https://issues.apache.org/jira/browse/CALCITE-4496). I added a note.